src/storage/storage.cpp

```diff
@@ -12,29 +12,194 @@
  * Public
  **********************************************************************************************************************/
 
+constexpr const char* kStorageInstancePath = CONFIG_AOS_STORAGE_PATH "/database_instance";
+constexpr const char* kStorageServicePath = CONFIG_AOS_STORAGE_PATH "/database_service.db";
+
 aos::Error Storage::Init()
 {
     LOG_DBG() << "Initialize storage";
 
+    aos::Error err;
+
+    aos::StaticString<aos::cFilePathLen> path {kStorageInstancePath};
+
+    if (!(err = mInstanceDatabase.Init(path)).IsNone()) {
+        return err;
+    }
+
+    path = kStorageServicePath;
+    if (!(err = mServiceDatabase.Init(path)).IsNone()) {
+        return err;
+    }
+
     return aos::ErrorEnum::eNone;
 }
 
 aos::Error Storage::AddInstance(const aos::InstanceInfo& instance)
 {
-    return aos::ErrorEnum::eNone;
+    aos::LockGuard lock(mMutex);
+
+    return mInstanceDatabase.Add(ConvertInstanceInfo(instance));
 }
 
 aos::Error Storage::UpdateInstance(const aos::InstanceInfo& instance)
 {
+    aos::LockGuard lock(mMutex);
+
+    return mInstanceDatabase.Update(ConvertInstanceInfo(instance), [&instance](const InstanceInfo& data) {
+        return data.mInstanceIdent.mInstance == instance.mInstanceIdent.mInstance
```

Comment:
I've added comparison function for instance ident in launcher implementation. After rebace you can compare instance ident directly.